src/main/kotlin/nl/avisi/structurizr/site/generatr/GenerateSiteCommand.kt

```diff
@@ -109,7 +109,12 @@ class GenerateSiteCommand : Subcommand(
             throw Exception("$defaultBranch does not contain a valid structurizr workspace. Site generation halted.")
         }
 
-        branchesToGenerate.forEach { branch ->
+        val sortedBranchNames = branchesToGenerate.toMutableList()
```

Comment:
We can do this more elegantly. My proposal:

Introduce a custom comparator for our needs on the top level of this file:

```kotlin
fun branchComparator(defaultBranch: String) = Comparator<String> { a, b ->
    if (a == defaultBranch) -1
    else if (b == defaultBranch) 1
    else a.compareTo(b)
}
```

The test for that one would look like:

```kotlin
class BranchComparatorTest {
    @TestFactory
    fun `default branch first then by string`() = listOf(
        listOf("a", "b", "main"),
        listOf("a", "main", "b"),
        listOf("b", "a", "main"),
        listOf("b", "main", "a"),
        listOf("main", "a", "b"),
        listOf("main", "b", "a")
    ).map { branches ->
        DynamicTest.dynamicTest(branches.toString()) {
            assertThat(branches.sortedWith(branchComparator("main")))
                .containsExactly("main", "a", "b")
        }
    }
}
```

Having this in place we can just add the following to line 104 to have `branchesToGenerate` already sorted on initialization:


```kotlin
}.sortedWith(branchComparator(defaultBranch))
```